### ISSUE TITLE
glitch PGM games on FBNEO libretro WII U

### ISSUE THREAD

**Issue by Zeox-Gamer** [finalburnneo/FBNeo#1780]
The games run well in terms of fps rate, however, all PGM games on FBNEO on WII U contain glitchy graphics in the background of cutscenes, making platformers like Demon Front unplayable due to not knowing where to jump on the platforms.
In all games the character sprites and life and energy bars are perfect, only the background graphics of the scenes are flawed, such as the ground and objects and landscape backgrounds, some of which are many in other games, not so much but they are incorrect.

**Comment by dinkc64** [comment]
Hi, since we don't have one of these devices to test on, please take some pictures of the bad scenes that you mentioned and post them here, so we can try to think about where the problem might be

best regards,
- dink

**Comment by barbudreadmon** [comment]
This one was reported last year (https://github.com/finalburnneo/FBNeo/commit/c09901b57c56116092ed6e6f31852c416987c2b6), i made some attempts at fixing it back then but without success.

@crystalct any chance you could take a look ?

**Comment by crystalct** [comment]
Just tested Demon Front .... it works perfectly... and ... a very nice game... :P
So on PS3 not glitchy graphics in the background.

**Comment by barbudreadmon** [comment]
@crystalct hmmm are you sure your ps3 build is using our latest pgm driver ? it got updated sometime last year

**Comment by crystalct** [comment]
this is a big problem... on the PS3 the current version of FBneo can no longer be compiled.. I should take the latest compileable version and manually make the changes to the new pgm driver

**Comment by barbudreadmon** [comment]
>on the PS3 the current version of FBneo can no longer be compiled.. 

Oh, is that anything we could help with ?

**Comment by crystalct** [comment]
no... too many things now refer to a more modern gcc.... we would need a working version made on PSL1GHT.... which has a decently modern version of gcc

**Comment by barbudreadmon** [comment]
>too many things now refer to a more modern gcc...

Do you have examples ? I was almost sure our codebase was still compatible with older gcc

**Comment by crystalct** [comment]
are you sure there aren't many more things that need g++ now?
Not so much fbneo core... but RETROARCH layer....

**Comment by barbudreadmon** [comment]
I don't think FBNeo should fail building on older gcc, but maybe retroarch does ?

**Comment by barbudreadmon** [comment]
Shit, so they entirely dropped support for ps3 after that sdk shitstorm a few years ago ?
Maybe current FBNeo core is buildable with older retroarch ?

**Comment by crystalct** [comment]
Absolutly no.... we need someone with a lot of free time... many many free time, to finish the porting of Retroarch on PSL1GHT

**Comment by Zeox-Gamer** [comment]
> Olá, como não temos um desses dispositivos para testar, tire algumas fotos das cenas ruins que você mencionou e poste aqui, para que possamos tentar pensar onde pode estar o problema
> 
> atenciosamente,
> 
>     * pensar

This error in PGM games on FBNEO core WII U
![demon front PGM GAME FBNEO wii u](https://github.com/finalburnneo/FBNeo/assets/169610613/7031de7f-cbf3-40bb-8a66-9f7be801d67a)
![spectral vs generation PGM GAME FBNEO wii u](https://github.com/finalburnneo/FBNeo/assets/169610613/ba9e9686-deb7-440b-9f48-18b5e0a68546)

**Comment by barbudreadmon** [comment]
some additional note : back when the issue was reported on discord, it was mentioned that other stages in demon front's attract mode didn't seem affected by the "missing floor" issue

**Comment by dinkc64** [comment]
hmm, so bg layer (nburnlayer & 1) is missing

**Comment by dinkc64** [comment]
@barbudreadmon, please merge the pgm_draw update, it's just a test to see if I need to dig deeper or not :)

**Comment by vaguerant** [comment]
Confirming no major changes since a year ago on Wii U, running d64b24c. Attached are some screenshots of the various issues occurring in Demon Front, or at least *where* they occur.

Missing floor on Stage 1 (Valley):
![dmnfrnt-240517-001732](https://github.com/finalburnneo/FBNeo/assets/5259025/1adb3228-0c46-4bad-84f2-f3b4c2ea5c92)
I don't remember noticing this last time, but there are screen-shaking explosion effects throughout this area and they result in that middle line of corrupted graphics at the vertical center of the screen.

Attract mode text crawl:
![dmnfrnt-240517-002322](https://github.com/finalburnneo/FBNeo/assets/5259025/a9139bb1-2b4f-4ac3-8db6-224260bd6980)
The problem here is not visible in a screenshot, but the text scrolls very awkwardly on Wii U, jumping up and down while it scrolls instead of smoothly moving up.

This screen runs if you leave the attract mode running for a bit. It goes:
* gameplay demo 1 (Valley)
    first screenshot
* fancy intro animation showing off the different characters
* gameplay demo 2 (Passage)
    notably, the floor works fine on this and other stages, only Valley has issues as far as I can see from the attract sequence
* text crawl
    second screenshot

Just to be clear, the issues are not limited to the attract mode. The Valley stage is missing the floor, etc. Here's some gameplay screenshots:

Station level (left on the stage select):
![dmnfrnt-240517-003657](https://github.com/finalburnneo/FBNeo/assets/5259025/008d8e40-e284-4a04-ae97-517383954328)
Background is missing in the opening area. The floor is here, though.

Forest level (right on the stage select):
![dmnfrnt-240517-003734](https://github.com/finalburnneo/FBNeo/assets/5259025/c418b8f3-8bcc-4e21-9f22-ee0911650262)
Might be fine? Definitely the least problematic of the first three stages which are unlocked in a new game.

**Comment by Zeox-Gamer** [comment]
> @Zeox-Gameré melhor com o novo núcleo?
I haven't tested it yet, can I update it via the Wiiu retroarch?

**Comment by Zeox-Gamer** [comment]
@barbudreadmon  friend, I updated the core yesterday using the Wii U retroarch and tested the PGM games, there was no difference, the errors continue

Unless the emulator you corrected is to be downloaded elsewhere, in that case I don't know where to download it other than through Retroarch Wiiu itself, like I did yesterday

**Comment by Zeox-Gamer** [comment]
Good friends, the error I reported still persists so I know if what I downloaded from retroarch is your corrected version or you haven't even touched it yet and I misunderstood, sorry if that's it
I have already done my part to report the error that occurred only in PGM games.

**Comment by Zeox-Gamer** [comment]
@barbudreadmon No friend, the latest version of Final Burn Neo on Wii U still contains the same errors.
I updated and tested it these days, it's exactly the same, the same defects as before in the pgm games.

**Comment by barbudreadmon** [comment]
Due to FBNeo hitting some kind of size limitations with libretro's wiiu buildbot last month, the FBNeo wiiu core can't be not updated anymore. I don't know when/if this buildbot problem will ever be fixed.

At this point, there isn't be much point in fixing big-endian issues since neither ps3 nor wiiu builds are updated anymore, so i can only recommend using a better device for your emulation needs, one that isn't underpowered and using some long dead big-endian architecture.

**Comment by dinkc64** [comment]
barbudreadmon, could you have the wiiu core compile without d_nes for a possible fix for build?

**Comment by vaguerant** [comment]
FYI, the buildbot is likely to be fixed in the (relatively) short term. There is ongoing work to update RetroArch for Wii U to run under the modern Aroma homebrew environment instead of the old, outdated way that it runs currently. Most other homebrew has already been switched over to the new environment. RetroArch is the one major holdout, as it is compiled with an older toolchain which cannot build for Aroma.

Downstream at [ashquarky/RetroArch](https://github.com/ashquarky/RetroArch/tree/wiiu-wut-merge), RetroArch has been updated to compile with the current toolchain, allowing it to build again on up-to-date consoles. This has also allowed current FBNeo to build and run without issue. Once all the kinks are ironed out and everything has been tested, these changes will be merged upstream and RetroArch for legacy environments will be sunsetted in favor of the modern (working) Aroma version.

Wii U's not dead yet! :P

**Comment by barbudreadmon** [comment]
@dinkc64 removing drivers would only be a short term solution, and i'm not entirely convinced removing only 1 driver would solve anything. i'd be more in favor with providing "several smaller cores" (as @crystalct did for its ps3 builds ?) but that'd require time and i don't have a lot of that atm.

@vaguerant yeah, i heard about that aroma thing that has been in the works for many years, but it doesn't seem there is any ETA yet for its use in libretro's buildbot.

**Comment by barbudreadmon** [comment]
>Very recently (in the last couple of weeks), the subject was brought up with libretro again and this time they were willing to just sunset the legacy version

The timing seems to match so my guess is that they freaked out when FBNeo stopped building.

**Comment by barbudreadmon** [comment]
@vaguerant any news about that aroma thingy ?
